ansible_events_ui/db/utils/lostream.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+import logging
+from io import UnsupportedOperation
+from typing import Tuple, Union
+
+from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui.db.dependency import get_db_session
+
+LOG = logging.getLogger(__name__)
+
+
+# Default read buffer size is
+CHUNK_SIZE = 1024**2
+
+
+class LObject:
+    """
+    Facilitate PostgreSQL large object interface using server-side functions.
+
+    As of 2022-09-01, there is no large object support directly
+    in asyncpg or other async Python PostgreSQL drivers.
+    """
+
+    VALID_MODES = {"rwt", "rt", "wt", "at", "rwb", "rb", "wb", "ab"}
+    INV_READ = 0x00040000
+    INV_WRITE = 0x00020000
+    MODE_MAP = {"r": INV_READ, "w": INV_WRITE, "a": INV_READ | INV_WRITE}
+
+    def __init__(
+        self,
+        oid: int = 0,
+        length: int = 0,
+        session: AsyncSession = None,
```

Comment:
The database session is essential for the class to work, so why is the `session` parameter optional?

P.S. If the session were optional, the correct annotation is `Optional[AsyncSession]`.

Reply:
Oops.